[CORL-2467] Make mobile toolbar's `Unmark all` work with server side seen comments

nick-funk opened the pull request:
## What does this PR do?

- Removes the giant indexed db seen comments provider
- Hooks all the old removed functionality into the new seen comments mutations/queries
- Updates the mutations for marking seen comments to handle multiple comments
- Hooks the keyboard and mobile toolbar actions into the updated mutations

## What changes to the GraphQL/Database Schema does this PR introduce?

Updates `MarkCommentSeenPayload` and `MarkCommentSeenInput` to handle multiple comments/id's.

## Does this PR introduce any new environment variables or feature flags? 

No.

## How do I test this PR?

- Enable the comment seen functionality.
- Setup your  environment to test on mobile (or deploy to a test environment on GKE)
- Make some comments with one user
- On your phone, view that comment stream as a different user
- See that the comments are yellow for un-seen
- Press `Unmark all` in the toolbar
- Comments should be marked seen (go white instead of yellow)
- Test other z-key functionality to make sure it isn't now broken from the changes
 
## How do we deploy this PR?

No special considerations.

nick-funk committed:
Remove seen context provider and update hooks

There was anther chunk of indexed db seen state
that needed to be removed and hooked into the
new server persisted seen state.

nick-funk committed:
Make unmark all work with new server side persisted seen state

nick-funk committed:
Remove un-necessary indented comment class

nick-funk committed:
Update snapshots

nick-funk committed:
Update ReplyList tests to work with useLocal

nick-funk commented on src/core/client/stream/tabs/Comments/ReplyList/ReplyList.spec.tsx:
```diff
@@ -1,42 +1,70 @@
+//   };
+//   const wrapper = shallow(<ReplyListN {...props} />);
+//   expect(wrapper).toMatchSnapshot();
+// });
```
I'm not rewriting this whole test into an integration test right now. We want to get rid of all these `shallow` render test anyways as they're very brittle and break any time any local state or relay changes happen due to the fact that all fragments are removed from the props when we create `ReplyListN` for this test.

kabeaty replied:
Sounds good, and thanks for leaving the comment with it.

nick-funk replied:
Actually, I found a way to fix it, test is updated and passing!

kabeaty commented on src/core/client/stream/tabs/Comments/Comment/CommentContainer.css:
```diff
@@ -73,12 +73,9 @@ $commenterActionEditColorActive: var(--palette-primary-300);
 }
 
-.indentedComment {
+.commentSeenEnabled {
```
It looks like the padding is missing again for comments in the logged-out state with these and related changes in the `CommentContainer`

nick-funk replied:
Hmm... I reverted this cause I found Vinh had done the padding elsewhere in the flow. I will look into it and see if I reversed the check for the classname or something.

nick-funk replied:
Yeah, putting the padding back on the CommentContainer, fixed, snapshot updates incoming!

kabeaty commented on src/core/client/stream/tabs/Comments/Comment/MarkCommentAsSeenMutation.tsx:
```diff
@@ -12,15 +12,12 @@ import {
 
 const mutation = graphql`
   mutation MarkCommentAsSeenMutation($input: MarkCommentSeenInput!) {
     markCommentAsSeen(input: $input) {
```
I'm wondering if it would make sense for this to be `markCommentsAsSeen` now?

nick-funk replied:
Oh dang! Yes! I even had that in a stash, but reverted to fix a different thing! I will rename it!

nick-funk replied:
All fixed!

kabeaty commented on src/core/client/stream/tabs/Comments/Comment/MarkCommentAsSeenMutation.tsx:
```diff
@@ -46,24 +43,28 @@ const enhanced = createMutation(
-            id: input.commentID,
-            seen: true,
-          },
+          comments: [
```
Should the optimistic response include all comments provided in the input marked as seen?

nick-funk replied:
Yup! Fixed this up!

kabeaty reviewed (changes requested):
Looking good! Manual testing confirms that it fixes `unmark all` and other shortcuts also still work as expected. I left a few comments throughout.

nick-funk committed:
Map comment ids for a better optimistic response

nick-funk committed:
Dummy test so the rest of ReplyList.spec.ts can run properly

nick-funk committed:
Add a mocked context to ReplyList.spec.tsx

nick-funk committed:
Rename MarkCommentAsSeen to MarkCommentsAsSeen

nick-funk committed:
Put padding back on indented comment

nick-funk committed:
Update snapshots

kabeaty commented on src/core/client/stream/tabs/Comments/Comment/MarkCommentAsSeenMutation.tsx:
```diff
@@ -9,18 +9,15 @@ import {
-  mutation MarkCommentAsSeenMutation($input: MarkCommentSeenInput!) {
-    markCommentAsSeen(input: $input) {
-      comment {
+  mutation MarkCommentAsSeenMutation($input: MarkCommentsAsSeenInput!) {
```
Maybe rename the mutation as well?

nick-funk replied:
Yep yep, didn't realize my code-wide rename was missing this. It's done now.

kabeaty commented on src/core/client/stream/tabs/Comments/ReplyList/__snapshots__/ReplyList.spec.tsx.snap:
```diff
@@ -62,163 +62,3 @@ exports[`renders correctly 1`] = `
   />
 </ForwardRef(forwardRef)>
 `;
-
```
Why are these snapshots removed? Looks like this part of the ReplyList tests has errors when run now.

nick-funk replied:
I honestly have no idea what is going on... I updated the tests, it worked, all was green, I pushed. Checking now.

nick-funk replied:
So weird, I ran all tests, all green, passed on my end. Then I removed initLocalState, ran all tests again, passed, all green. No changes to snapshots. So I then forced it with `npm run test ReplyList.spec` to tell it to just run the `ReplyList` tests and two snapshots popped out. This is super flaky and weird. Hopefully this fixes it now though.

nick-funk replied:
Okay, sorted, it's all good now.

kabeaty reviewed (commented):
Thanks for making the changes! I left a few more comments about potential renaming for consistency and some test errors that I'm seeing now.

nick-funk committed:
Rename MarkCommentAsSeenMutation -> MarkCommentsAsSeenMutation

nick-funk committed:
Remove initLocalState to let useContext populate Local

For some reason this was passing? Even though the snap was
missing? It is now also passing, but I don't know why. This
makes no sense.

Pushing this up and hoping it makes the CirclCI builds happy
as well.

nick-funk committed:
Sort imports

nick-funk committed:
Fix up import namings that VSCode bungled
